crates/scheduler/src/cfs.rs

```diff
@@ -0,0 +1,173 @@
+use alloc::{collections::BTreeMap, sync::Arc};
+use core::ops::Deref;
+use core::sync::atomic::{AtomicIsize, Ordering};
+
+use crate::BaseScheduler;
+
+/// task for CFS
+pub struct CFTask<T> {
+    inner: T,
+    init_vruntime: AtomicIsize,
+    delta: AtomicIsize,
+    nice: AtomicIsize,
+    id: AtomicIsize,
+}
+
+// TODO：现在全部都是暴力实现
+
+const NICE2WEIGHT_POS: [isize; 20] = [
+    1024, 820, 655, 526, 423, 335, 272, 215, 172, 137, 110, 87, 70, 56, 45, 36, 29, 23, 18, 15,
+];
+const NICE2WEIGHT_NEG: [isize; 21] = [
+    1024, 1277, 1586, 1991, 2501, 3121, 3906, 4904, 6100, 7620, 9548, 11916, 14949, 18705, 23254,
+    29154, 36291, 46273, 56483, 71755, 88761,
+];
+
+impl<T> CFTask<T> {
+    /// new with default values
+    pub const fn new(inner: T) -> Self {
+        Self {
+            inner,
+            init_vruntime: AtomicIsize::new(0_isize),
+            delta: AtomicIsize::new(0_isize),
+            nice: AtomicIsize::new(0_isize),
+            id: AtomicIsize::new(0_isize),
+        }
+    }
+
+    fn get_weight(&self) -> isize {
+        if self.nice.load(Ordering::Acquire) >= 0 {
+            NICE2WEIGHT_POS[self.nice.load(Ordering::Acquire) as usize]
+        } else {
+            NICE2WEIGHT_NEG[(-self.nice.load(Ordering::Acquire)) as usize]
+        }
+    }
+
+    fn get_id(&self) -> isize {
+        self.id.load(Ordering::Acquire)
+    }
+
+    fn get_vruntime(&self) -> isize {
+        if self.nice.load(Ordering::Acquire) == 0 {
+            self.init_vruntime.load(Ordering::Acquire) + self.delta.load(Ordering::Acquire)
+        } else {
+            self.init_vruntime.load(Ordering::Acquire)
+                + self.delta.load(Ordering::Acquire) * 1024 / self.get_weight()
+        }
+    }
+
+    fn set_vruntime(&self, v: isize) {
+        self.init_vruntime.store(v, Ordering::Release);
+    }
+
+    fn set_id(&self, id: isize) {
+        self.id.store(id, Ordering::Release);
+    }
+
+    fn task_tick(&self) -> isize {
+        self.delta.fetch_add(1, Ordering::Release);
+        self.get_vruntime()
```

Comment:
`self.get_vruntime()` has no effect?

Reply:
I've already fix this, but it still remains unresolved?